CocktailRecipes/CocktailRecipes/Networking/Core/NetworkManager.swift

```diff
@@ -84,22 +68,14 @@ final class NetworkManager {
 		case .request:
 			request.setValue("application/json", forHTTPHeaderField: "Content-Type")
 		case.requestParameters(let parameters):
-			do {
-				try encodeParameter(parameters: parameters, with: &request)
-			} catch {
-				throw error
-			}
+			try encodeParameter(parameters: parameters, with: &request)
 		}
 
 		return request
 	}
 
 	private func encodeParameter(parameters: Parameters, with request: inout URLRequest) throws {
 		let parameterEncoder = URLParameterEncoder()
-		do {
-			try parameterEncoder.encode(urlRequest: &request, with: parameters)
-		} catch {
-			throw error
-		}
+        try parameterEncoder.encode(urlRequest: &request, with: parameters)
```

Comment:
who would catch the `paremeterEncodingFailed" error, if its not done here?

Reply:
The function is throwing already. Any thing that throws inside this function will be forwarded.

Reply:
perfect